### PR TITLE
Enhancement to bash scripts

### PR DESCRIPTION
**Pull request by BShraman**
Added return code check to minimize writing to txt file in every loop. 

**This request is just to review below changes for now** . 
I have added two different version for return code check logic. Both of those should produce the similar results . 
V1 : This is recommended code , but wont be able to print return code from the command line.  
V2 : Eval function should be avoided , as it might manipulate the input . But returned code can be printed in log file. 

Any thoughts ?  I will be resubmitting clean merge request, once we agreed on the logic. 

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Review by o-nikolas — COMMENTED** [review]
> V1 : This is recommended code , but wont be able to print return code from the command line.

I don't believe this is true. You can easily grab the return code without using `eval`:

In fact you don't even need to put the command into a variable. Here is a test script that you can play with:

```
while echo "running"; do
    ls /tmp/test_file > /dev/null
    exit_code=$?
    if (( exit_code != 0 ))
    then
        echo "error listing test file: $exit_code. Recreating it..." >&2
        touch /tmp/test_file
    fi
    sleep 1
done
```
The "process" here is just a `ls` command, listing a test file. If the return code is zero the watchdog just runs again silently, if it's a non-zero return code then it, logs, repairs the file and then loops again. It **successfully logs the return code as well**, like you're hoping. Here is an example execution where I delete the test file halfway through to simulate the failure:
```
niko:/tmp$ ./myscript.sh 
running
running
running
running
running
ls: cannot access '/tmp/test_file': No such file or directory
error listing test file: 2. Recreating it...
running
running
^C 
```

### What's the goal here though?

I'd like to hear more details about what exactly you're trying to achieve here. You seem to be looking for cases where the scheduler is exiting gracefully with a zero exit code and you'd like to **not** log or sleep in those cases. Is that right? I wonder if the process exits gracefully in this way if we should be restarting it at all? Or perhaps in that case the user truly wanted the process to stop.

**Comment by BShraman** [comment]
Thank you @o-nikolas for your feedback. I have tested as you mentioned and script is working as expected. I have requested clean version of code to be merge again. 

Main focus of enhancement is to add return code check and write to file when scheduler crash.

**Review by uranusjr — APPROVED** [review]

**Comment by uranusjr** [comment]
Approving in the sense that it should achieve what you want it to, but I'd also like some clarification on what the exact goal is here, as mentioned by @o-nikolas.

**Comment by potiuk** [comment]
> I'd like to hear more details about what exactly you're trying to achieve here. You seem to be looking for cases where the scheduler is exiting gracefully with a zero exit code and you'd like to **not** log or sleep in those cases. Is that right? I wonder if the process exits gracefully in this way if we should be restarting it at all? Or perhaps in that case the user truly wanted the process to stop.

I agree here with @o-nikolas - I think it would be great also to exit when scheduler exits gracefully. I am not sure, however if there is such case altogether. Usually that should be done when scheduler gets TERM signal and manages to complete everything and quit. Then it should not be restarted. In fact the purpose of the script is also unclear to me. It is part of the codebase, but it's neither used nor mentioned anywhere in the documentation. So if we are changing it, I guess we should also mention how it should be used  (in charts? In docker compose? elsewhere?), why we cannot use the existing deployment mechanisms for that and what is the characteristics of this script - i.e. should it really run in a loop regardles of the exit code?

File: INTHEWILD.md [hunk 1]
```diff
@@ -397,6 +397,7 @@ Currently, **officially** using Airflow:
 1. [Umami Collective](https://umamicollective.com) [[@juanuicich](https://github.com/juanuicich)]
 1. [United Airlines](https://www.united.com/) [[@ilopezfr](https://github.com/ilopezfr)]
 1. [Upsight](https://www.upsight.com)
+1. [US Bank](https://www.usbank.com)[@BShraman]
```
Comment (potiuk) [inline INTHEWILD.md]: Should be  a separate PR

Comment (BShraman) [reply INTHEWILD.md]: My bad, i am trying to pull back request "INTHEWILD.md" , is there is a way ??

Comment (potiuk) [reply INTHEWILD.md]: Rebase to latest main, amend the commit and push. If you are not familiar with rebase workflow see https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#how-to-rebase-pr

Comment (BShraman) [reply INTHEWILD.md]: Done  and requested ... Thank you @potiuk.

**Review by potiuk — APPROVED** [review]

**Pull request merged**